mock/src/test_ctx.rs

```diff
@@ -115,74 +113,21 @@ impl<const NACC: usize, const NTX: usize> TestContext<NACC, NTX> {
         Fb: FnOnce(&mut MockBlock, Vec<MockTransaction>) -> &mut MockBlock,
         FAcc: FnOnce([&mut MockAccount; NACC]),
     {
-        let mut accounts: Vec<MockAccount> = vec![MockAccount::default(); NACC];
-        // Build Accounts modifiers
-        let account_refs = accounts
-            .iter_mut()
-            .collect_vec()
-            .try_into()
-            .expect("Mismatched len err");
-        acc_fns(account_refs);
-        let accounts: [MockAccount; NACC] = accounts
-            .iter_mut()
-            .map(|acc| acc.build())
-            .collect_vec()
-            .try_into()
-            .expect("Mismatched acc len");
-
-        let mut transactions = vec![MockTransaction::default(); NTX];
-        let tx_refs = transactions.iter_mut().collect();
-
-        // Build Tx modifiers.
-        func_tx(tx_refs, accounts.clone());
-
-        // Sets the transaction_idx and nonce after building the tx modifiers. Hence, if user has
-        // overridden these values above using the tx modifiers, that will be ignored.
-        let mut acc_tx_count = vec![0u64; NACC];
-        transactions.iter_mut().enumerate().for_each(|(idx, tx)| {
-            let idx = u64::try_from(idx).expect("Unexpected idx conversion error");
-            tx.transaction_idx(idx);
-            if let Some((pos, from_acc)) = accounts
-                .iter()
-                .find_position(|acc| acc.address == tx.from.address())
-            {
-                tx.nonce(from_acc.nonce + acc_tx_count[pos]);
-                acc_tx_count[pos] += 1;
-            }
-        });
-
-        let transactions: Vec<MockTransaction> =
-            transactions.iter_mut().map(|tx| tx.build()).collect();
-
-        // Build Block modifiers
-        let mut block = MockBlock::default();
-        block.transactions.extend_from_slice(&transactions);
-        func_block(&mut block, transactions).build();
-
-        let chain_id = block.chain_id;
-        let block = Block::<Transaction>::from(block);
-        let accounts: [Account; NACC] = accounts
-            .iter()
-            .cloned()
-            .map(Account::from)
-            .collect_vec()
-            .try_into()
-            .expect("Mismatched acc len");
-
-        let geth_traces = gen_geth_traces(
-            chain_id,
-            block.clone(),
-            accounts.to_vec(),
-            history_hashes.clone(),
+        let test_ctx2 = TestContext2::<NACC, NTX, 0>::new_with_logger_config(
```

Comment:
We need one more Generic argument for the number of withdrawals, `NWD` (the 3rd Generic argument in`TestContext2<NACC, NTX, NWD>`). If we just used `TestContext`, a lot of test files will be changed. That's why I introduced `TestContext2`. 

I'm not sure how to avoid files changing by adding "`func_wd: FWd` in original TestContext api `new`/`new_with_logger_config`". Could you share some ideas to me?

Reply:
Yeah it requires lots of files change for many place use TestContext. We can do it in next PR
Besides, I think we can move generic from struct to method level, it achieved the same goal for assuring the type of functional closure, also we can add different factory methods to create TextContext without specifying some default generic, e.g. 0.

An example, FYI https://github.com/privacy-scaling-explorations/zkevm-circuits/compare/main...hero78119:test_ctx_refactor?expand=1

Reply:
issue created https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1651 and added comment in [d616eda](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1634/commits/d616eda437f3316ec3468525fea605975b801897)